src/gentables/CMakeLists.txt

```diff
@@ -2,13 +2,13 @@ cmake_minimum_required(VERSION 3.5)
 
 # remove $CC from the current environment and by that force cmake to look for a (working) C compiler,
 # which hopefully will be the host compiler
-unset(ENV{CC})
+set(ENV{CC} "$ENV{CC_FOR_BUILD}")
 
 # also unset $CFLAGS to avoid passing any cross compilation flags to the host compiler
-unset(ENV{CFLAGS})
+set(ENV{CFLAGS} "$ENV{CFLAGS_FOR_BUILD} $ENV{CPPFLAGS_FOR_BUILD}")
```

Comment:
How about combining them to one env var?

Reply:
CMake is infamous for ignoring the CPPFLAGS variable. However, some cross building environments assign separate parameters to CFLAGS and CPPFLAGS, because that is natively supported by Autotools. I'd like to assign both to CFLAGS at once and be explicit about this.